samples/msal-node-samples/ElectronSystemBrowserApp/README.md

```diff
@@ -0,0 +1,217 @@
+
+```typescript
+private static setDefaultProtocol(): void {
+        if (process.defaultApp) {
+            if (process.argv.length >= 2) {
+                Main.application.setAsDefaultProtocolClient(
+                    authConfig.customProtocol.name, // e.g msal4b0db8c2-9f26-4417-8bde-3f0e3656f8e
+                    process.execPath,
+                    [path.resolve(process.argv[1])]
+                );
+            } else {
+                Main.application.setAsDefaultProtocolClient(
+                    authConfig.customProtocol.name // e.g msal4b0db8c2-9f26-4417-8bde-3f0e3656f8e
+                );
+            }
+        }
+    }
+```
+
+Next, we tell the application how to handle events that invoke the application instance (such as navigation to the custom url scheme in browser). This code will differ between OSs.
+
+#### In Windows
+
+When the external custom URL scheme is clicked it will trigger the event `Main.application.on("second-instance", Main.onSecondInstance);` where we can fetch the authorization code from custom URL scheme:
+
+```typescript
+ private static onSecondInstance(
+        event: any,
+        commandLine: any,
+        workingDirectory: any
+    ): void {
+        // Someone tried to run a second instance, we should focus our window.
+        if (Main.mainWindow) {
+            if (Main.mainWindow.isMinimized()) Main.mainWindow.restore();
+            Main.mainWindow.focus();
+        }
+
+        //get the deep linked from command line params
+        const url = Main.getDeepLinkUrl(commandLine);
+
+        if (url) {
+            Main.getCodeFromDeepLinkUrl(url);
+        }
+    }
+```
+
+#### MacOS and Linux
+
+When the external custom URL scheme is clicked it will trigger the event `Main.application.on("open-url", Main.onOpenUrl);` where we can fetch the authorization code from custom URL scheme:
+
+```typescript
+ private static onOpenUrl(event: any, schemeData: string) {
+        event.preventDefault();
+        // Someone tried to run a second instance, we should focus our window.
+        if(Main.mainWindow){
+            if (Main.mainWindow.isMinimized()) Main.mainWindow.restore();
+            Main.mainWindow.focus();
+        }
+
+        //get the deep linked from command line params
+        const url = Main.getDeepLinkUrl(schemeData);
+
+         if (url) {
+             Main.getCodeFromDeepLinkUrl(url);
+         }
+
+    }
+```
+
+**Note: On macOS and Linux, this feature will only work when your app is packaged. It will not work when you're launching it in development from the command-line. For more information on how to pack an electron application please check the [electron documentation](https://www.electronjs.org/docs/latest/tutorial/launch-app-from-url-in-another-app#packaging>).**
```

Comment:
@salman90 can you confirm this? I was able to run it directly on my macOS : )

Reply:
In my case, I had to pack the application, But I updated the note's description to "You May need to pack the app."  just in case it worked or didn't work in development.